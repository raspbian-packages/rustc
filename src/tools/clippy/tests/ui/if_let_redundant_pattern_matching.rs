#![feature(plugin)]

#![plugin(clippy)]
#![warn(clippy)]
#![warn(if_let_redundant_pattern_matching)]


fn main() {
    if let Ok(_) = Ok::<i32, i32>(42) {}

    if let Err(_) = Err::<i32, i32>(42) {
    }

    if let None = None::<()> {
    }

    if let Some(_) = Some(42) {
    }

    if Ok::<i32, i32>(42).is_ok() {
    }

    if Err::<i32, i32>(42).is_err() {
    }

    if None::<i32>.is_none() {
    }

    if Some(42).is_some() {
    }

    if let Ok(x) = Ok::<i32,i32>(42) {
        println!("{}", x);
    }
}
