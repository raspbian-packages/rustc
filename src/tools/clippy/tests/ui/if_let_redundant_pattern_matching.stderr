error: redundant pattern matching, consider using `is_ok()`
 --> $DIR/if_let_redundant_pattern_matching.rs:9:12
  |
9 |     if let Ok(_) = Ok::<i32, i32>(42) {}
  |     -------^^^^^--------------------- help: try this: `if Ok::<i32, i32>(42).is_ok()`
  |
  = note: `-D if-let-redundant-pattern-matching` implied by `-D warnings`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/if_let_redundant_pattern_matching.rs:11:12
   |
11 |     if let Err(_) = Err::<i32, i32>(42) {
   |     -------^^^^^^---------------------- help: try this: `if Err::<i32, i32>(42).is_err()`

error: redundant pattern matching, consider using `is_none()`
  --> $DIR/if_let_redundant_pattern_matching.rs:14:12
   |
14 |     if let None = None::<()> {
   |     -------^^^^------------- help: try this: `if None::<()>.is_none()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/if_let_redundant_pattern_matching.rs:17:12
   |
17 |     if let Some(_) = Some(42) {
   |     -------^^^^^^^----------- help: try this: `if Some(42).is_some()`

error: aborting due to 4 previous errors

